src/main/java/org/myrobotlab/service/BodyPart.java

```diff
@@ -219,4 +226,108 @@ static public ServiceType getMetaData() {
     meta.addDependency("org.apache.commons", "commons-lang3", "3.3.2");
     return meta;
   }
+
+  //pasted from inmoov service, for compatibility
+  @Override
+  public void onJointAngles(Map<String, Double> angleMap) {
+    // We should walk though our list of servos and see if
+    // the map has it.. if so .. move to it!
+    // Peers p = InMoovArm.getPeers(getName()).getPeers("Servo");
+    // TODO: look up the mapping for all the servos in the arm.
+
+    // we map the servo 90 degrees to be 0 degrees.
+    HashMap<String, Double> phaseShiftMap = new HashMap<String, Double>();
+    // phaseShiftMap.put("omoplate", 90);
+    // Harry's omoplate is +90 degrees from Gaels InMoov..
+    // These are for the encoder offsets.
+    // these map between the reference frames of the dh model & the actual arm.
+    // (calibration)
+    phaseShiftMap.put("omoplate", 90.0);
+    phaseShiftMap.put("shoulder", 90.0);
+    phaseShiftMap.put("rotate", -450.0);
+    phaseShiftMap.put("bicep", 90.0);
+
+    HashMap<String, Double> gainMap = new HashMap<String, Double>();
+    gainMap.put("omoplate", 1.0);
+    gainMap.put("shoulder", -1.0);
+    gainMap.put("rotate", -1.0);
+    gainMap.put("bicep", -1.0);
+
+    ArrayList<String> servos = new ArrayList<String>();
+    servos.add("omoplate");
+    servos.add("shoulder");
+    servos.add("rotate");
+    servos.add("bicep");
+    for (String s : servos) {
+      if (angleMap.containsKey(s)) {
+        if ("omoplate".equals(s)) {
+          Double angle = (gainMap.get(s) * angleMap.get(s) + phaseShiftMap.get(s)) % 360.0;
+          if (angle < 0) {
+            angle += 360;
+          }
+          getServo("omoplate").moveTo(angle.intValue());
+        }
+        if ("shoulder".equals(s)) {
+          Double angle = (gainMap.get(s) * angleMap.get(s) + phaseShiftMap.get(s)) % 360.0;
+          if (angle < 0) {
+            angle += 360;
+          }
+          getServo("shoulder").moveTo(angle.intValue());
+        }
+        if ("rotate".equals(s)) {
+          Double angle = (gainMap.get(s) * angleMap.get(s) + phaseShiftMap.get(s)) % 360.0;
+          if (angle < 0) {
+            angle += 360;
+          }
+          getServo("rotate").moveTo(angle.intValue());
+        }
+        if ("bicep".equals(s)) {
+          Double angle = (gainMap.get(s) * angleMap.get(s) + phaseShiftMap.get(s)) % 360.0;
+          getServo("bicep").moveTo(angle.intValue());
+          if (angle < 0) {
+            angle += 360;
+          }
+        }
+      }
+    }
+  }
+  
+  //pasted from inmoov service, for compatibility
+  public static DHRobotArm getDHRobotArm() {
+
+    // TODO: specify this correctly and document the reference frames!
+    DHRobotArm arm = new DHRobotArm();

```

Comment:
Similar, this is specific to the inmoov arm, and not a generic body part.  Therefore this should return null and log a warning message that it's not implemented.  (Another approach would be to make it abstract and force a subclass to implement it.